### PR TITLE
Working redux to QuestionsList

### PR DESCRIPTION
**Pull request by iwantmyhatback**

### CODE REVIEW

File: src/components/questions/question.js [hunk 1]
```diff
@@ -9,7 +9,24 @@ class Question extends React.Component{
   }
 
   render(){
-    return (<div><div>Question Content</div><AnswerList/></div>);
+    return (<div>
```
Comment (dprevoznik) [inline src/components/questions/question.js:12]: I like the way this is set up. Any idea on how you plan on eventually styling this component and what other boilerplate you will require?

File: src/components/questions/questionsList.js [hunk 2]
```diff
@@ -10,8 +10,21 @@ constructor(props){
...
+  console.log(this.props)
+  this.props.getProductQuestions(3);
+}
+
 render(){
-  return <div><div>QuestionList (map out questions here)</div><Question/></div>
+  return (
+  <div>
+    {
+    this.props.store.questions[0] ? (this.props.store.questions.map( (question, index) => {
```
Comment (dprevoznik) [inline src/components/questions/questionsList.js:22]: Good logic used here. Good idea we should use going forward on mapping only if the data is present.

File: src/containers/questions/questionListContainer.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import { connect } from "react-redux";
+import questionsList from "../../components/questions/questionsList";
+import { getProductQuestions } from "../../redux/actions/questionsActions/getQuestions";
+
+const mapStateToProps = (state, props)=>({store:state});
```
Comment (dprevoznik) [inline src/containers/questions/questionListContainer.js:5]: Did you plan on using props here? Interested in what you would be passing down to question list from above.

File: src/redux/actions/questionsActions/getQuestions.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+import { GET_QUESTIONS } from "../actionTypes.js";
+
+export function getProductQuestions(id) {
+  return function(dispatch) {
+    return axios
+      .get(`http://3.134.102.30/qa/${id}`)
+      .then(data => {
+        console.log("*** Sucessfully Get-ted Question Data From Server");
+        // console.log(data);
+        dispatch({
```
Comment (dprevoznik) [inline src/redux/actions/questionsActions/getQuestions.js:11]: Nice object literal use for inside dispatch. Are you dispatching this action numerous times? May be worth making a function that could be reused.

File: src/redux/actions/questionsActions/questionListCreator.js [hunk 5]
```diff
@@ -0,0 +1,7 @@
+import { GET_QUESTIONS } from "../actionTypes"
+
+export function getQuestions() {
+  return {
+    type: GET_QUESTIONS
```
Comment (dprevoznik) [inline src/redux/actions/questionsActions/questionListCreator.js:5]: I like that you didn't force a payload to be included here. It looks sharper and doesn't add unnecessary complexity.

**Review by dprevoznik — COMMENTED** [review]
Approved. I left notes on the code.

File: src/components/questions/question.js (hunk 1 above)
Comment (iwantmyhatback) [reply src/components/questions/question.js:12]: Im still laying out that styling on paper, i will make sure to make it available once finalized!

File: src/containers/questions/questionListContainer.js (hunk 3 above)
Comment (iwantmyhatback) [reply src/containers/questions/questionListContainer.js:5]: Honestly i just added that when i was was looking through YOUR code to remind myself that it was an option! Thats definitely something that might slip my mind at a later time but be handy to remember. Will probably remove it at some point once all my redux set up is finalized

File: src/redux/actions/questionsActions/getQuestions.js (hunk 4 above)
Comment (iwantmyhatback) [reply src/redux/actions/questionsActions/getQuestions.js:11]: I will look into that actually, i didn't consider that when constructing this request/dispatch

**Pull request merged**